prov/util/src/util_shm.c

```diff
@@ -82,6 +82,48 @@ void smr_cma_check(struct smr_region *smr, struct smr_region *peer_smr)
 	peer_smr->cma_cap = smr->cma_cap;
 }
 
+size_t smr_calculate_size_offsets(size_t tx_count, size_t rx_count,
+				  size_t *cmd_offset, size_t *resp_offset,
+				  size_t *inject_offset, size_t *sar_offset,
+				  size_t *peer_offset, size_t *name_offset)
+{
+	size_t cmd_queue_offset, resp_queue_offset, inject_pool_offset;
+	size_t sar_pool_offset, peer_data_offset, ep_name_offset;
+	size_t tx_size, rx_size, total_size;
+
+	tx_size = roundup_power_of_two(tx_count);
+	rx_size = roundup_power_of_two(rx_count);
+
+	cmd_queue_offset = sizeof(struct smr_region);
+	resp_queue_offset = cmd_queue_offset + sizeof(struct smr_cmd_queue) +
+			    sizeof(struct smr_cmd) * rx_size;
+	inject_pool_offset = resp_queue_offset + sizeof(struct smr_resp_queue) +
+			     sizeof(struct smr_resp) * tx_size;
+	sar_pool_offset = inject_pool_offset + sizeof(struct smr_inject_pool) +
+			  sizeof(struct smr_inject_pool_entry) * rx_size;
+	peer_data_offset = sar_pool_offset + sizeof(struct smr_sar_pool) +
+			   sizeof(struct smr_sar_pool_entry) * SMR_MAX_PEERS;
+	ep_name_offset = peer_data_offset + sizeof(struct smr_peer_data) * SMR_MAX_PEERS;
+
+	if (cmd_offset)
+		*cmd_offset = cmd_queue_offset;
+	if (resp_offset)
+		*resp_offset = resp_queue_offset;
+	if (inject_offset)
+		*inject_offset = inject_pool_offset;
+	if (sar_offset)
+		*sar_offset = sar_pool_offset;
+	if (peer_offset)
+		*peer_offset = peer_data_offset;
+	if (name_offset)
+		*name_offset = ep_name_offset;
+
+	total_size = ep_name_offset + NAME_MAX;
+	total_size = roundup_power_of_two(total_size);
```

Comment:
I know this is just copied from the old code, but is there a reason why the total size is rounded up to a power of two?

Reply:
That was a remnant of the code that I started from when I first started writing the provider. Something about making sure it was aligned? idk. I don't think it's necessary.

Reply:
I would just make a note to revisit this after the release then.  Maybe it doesn't need any size adjustment, or at most align to a multiple of a page size, rather than potentially almost doubling it.

Reply:
Let me remove it.

Reply:
I would keep it for now and make a note to look into it later like Sean suggests. I don't want to break anything for the release.

Reply:
Sure.